image_processing/src/views/layout_utils.py

```diff
@@ -0,0 +1,548 @@
+from h2o_wave import ui, main, Q, app
+import ip_utils as ip
+import os
+
+
+async def edge_detection(q: Q):
+    choices = [
+        ui.choice('laplace', 'Laplace'),
+        ui.choice('sobel', 'Sobel')
+    ]
+    if q.client.blur_controls:
```

Comment:
The blur_controls and histogram_match are checked in most of these function calls, it might be nice to put that all in one function you call, so that it's easier to make changes to clean up in the future